cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -269,7 +275,9 @@ public synchronized void updateProperties(Id.Namespace namespaceId, NamespaceMet
     if (config != null && !Strings.isNullOrEmpty(config.getSchedulerQueueName())) {
       builder.setSchedulerQueueName(config.getSchedulerQueueName());
     }
-
+    if (config != null && !Strings.isNullOrEmpty(config.getRootDirectory())) {
```

Comment:
We should also throw an exception if someone tries to remove the field.
For instance, if the value is empty string or null in the request, but currently is some custom mapping.
Right?